Source/DafnyCore/Resolver/CheckTypeCharacteristics_Visitor.cs

```diff
@@ -334,42 +334,50 @@ public CheckTypeCharacteristics_Visitor(ErrorReporter reporter)
     for (var i = 0; i < formalTypeArgs.Count; i++) {
       var formal = formalTypeArgs[i];
       var actual = actualTypeArgs[i];
-      if (!CheckCharacteristics(formal.Characteristics, actual, inGhostContext, out var whatIsWrong, out var hint)) {
-        reporter.Error(MessageSource.Resolver, tok, "type parameter{0} ({1}) passed to {2} {3} must support {4} (got {5}){6}",
-          actualTypeArgs.Count == 1 ? "" : " " + i, formal.Name, what, className, whatIsWrong, actual, hint);
+      if (!CheckCharacteristics(formal.Characteristics, actual, inGhostContext, out var whatIsNeeded, out var hint)) {
```

Comment:
Excellent philosophical renaming of a variable! Good error messages always indicate how to fix them, not that the user did wrong :-)